src/rgw/rgw_auth_s3.cc

```diff
@@ -408,6 +408,19 @@ static inline int parse_v4_auth_header(const req_info& info,               /* in
   }
   date = d;
 
+  uint64_t req_sec = (uint64_t)internal_timegm(&t);
```

Comment:
I think we should be using new ceph::time primitives here;  @adamemerson?

Reply:
hi @mattbenjamin, thanks for your comments.
I copied "internal_timegm" from function parse_v4_query_string. Would you tell me what "new ceph::time" is?

Reply:
i think the `internal_timegm()` bit is fine. there is a `common/ceph_time.h` header that's based on the `std::chrono` library that could be used instead of the `ceph_clock_now()` part. @adamemerson, do you see any advantages to using that here?

Reply:
@cbodley I'd prefer we use that going forward.

Reply:
Hi @adamemerson @mattbenjamin @cbodley,  do you mean change ceph_clock_now to ceph::real_clock::now?